code/move/solution/trymove.sol.cpp

```diff
@@ -1,39 +1,87 @@
-#include <iostream>
-#include <vector>
+#include <memory>
 #include <random>
-#include "NVector.sol.hpp"
-
-template<typename T>
-void swap(std::vector<T> &v, int a, int b) {
-    T mem = std::move(v[a]);
-    v[a] = std::move(v[b]);
-    v[b] = std::move(mem);
-    // even better
-    //swap(v[a], v[b]);
-}
+#include <vector>
 
-template<typename T>
-void randomize(std::vector<T> &v) {
-    // we randomize via len random inversions
-    std::size_t len = v.size();
-    std::default_random_engine e;
-    std::uniform_int_distribution<std::size_t> d{0, len};
-    for (std::size_t i = 0; i < len; i++) {
-        int a = d(e);
-        int b = d(e);
-        swap(v, a, b);
+/**
+ * An attempt to make a movable class. It allocates memory on construction, and has
+ * array semantics. (Prefer std::array or std::vector instead of hand-crafted solutions, though.)
+ * We made it copyable by providing copy constructor and assignment operator, but note that we
+ * are violating the rule of 5, so it's not moveable yet.
+*/
+class Moveable {
```

Comment:
Could we give this a more descriptive name? Like e.g. `DynArray`.

Reply:
I wanted to call it something with array first, but at the same time that might connect this things with real arrays in the minds of the students. We don't want that, do we?

Reply:
But your type is kind of array-ish, it has a size and a subscript :)

Reply:
I call it CustomArray now, to give it a bad connotation.